### ISSUE TITLE
Abillity to set elevation of Y/X axis.

### ISSUE THREAD

**Issue by NemyaNation** [Abhinandan-Kushwaha/react-native-gifted-charts#117]
Would be great if we could render our own text component for Y-axis, similarly to how we can for x-axis. Then we could personalize and style our own text strings completely, including it's container.

Failing that, if we can set the elevation, to "above"/"below" or 0/1 for x/y axis labels.

![graph-example2](https://user-images.githubusercontent.com/17049683/160626913-14999952-b3c5-4e1f-88b8-f651514d764c.png)

---

tldr: I want the ability to display my text labels above the curve and overlay.

**Comment by NemyaNation** [comment]
Hey @Abhinandan-Kushwaha thanks for the quick response.

I'm not entirely sure this is very useful for specific use cases revolving around the text component.

The best solution would be to allow for custom text components, similar to the xAxis current implementation.

---

As for the solution you propsed, this is still a problem for my original example, where the graph shadow as well as the graph line itself have a higher level than the yAxis labels.

![graph-example3](https://user-images.githubusercontent.com/17049683/160708417-bc09fee6-7be6-4fa6-afee-56c9638623ca.png)

^ that is using the new prop field (I updated the package thanks!)


Building up on this issue: It would be great if we could get a `yAxisLabelTextContainerStyle` meaning the container to be applied over each individual <Text> object. The reason for this is because then and only then can we set something like `textAlign: "flex-end"` on the container and be able to position the yAxis labels always aligned to the yAxis regardless of their size. meaning that if it's $100,000 or $1 it will always appear aligned to the yAxis.

Currently this is not possible since I cannot access the container of the individual text component. (which is needed since the text componenet is absolutely positioned so we can get it above the chart itself)

To explain this , look at this:
![graph-example3](https://user-images.githubusercontent.com/17049683/160709069-677fe748-9288-4f19-ac31-e96323e18666.png)

It would be good if we could access the container for each of the yAxis labels (or alternately just provide our own react native nodes)

**Comment by Abhinandan-Kushwaha** [comment]
Hi @NemyaNation 
I truly appreciate your idea of providing a custom component for the yAxis labels. But, at the moment, it's a bit hard to implement. This is because, currently the y axis labels are clubbed with the horizontal rules.

For that reason, I have launched another update- version `1.0.13` and it contains the prop `horizontalRulesStyle` which will solve your use case. Just add the prop `horizontalRulesStyle={{zIndex:1}}`

For the alignment part, try putting the textAlign property in the `yAxisTextStyle` or the `yAxisLabelContainerStyle` prop.
Also, there's a prop named `yAxisSide` which can be used here.

**Comment by NemyaNation** [comment]
Hey @Abhinandan-Kushwaha, have you tried using `horizontalRulesStyle={{zIndex:1}}` in test?

For me it's blacking out my entire graph, only the y-x axis labels are visible; and the graph completely dissapears.

**Comment by Abhinandan-Kushwaha** [comment]
Hi @NemyaNation 👋
Putting `horizontalRulesStyle={{zIndex:1}}` is blacking out the entire chart because you are using the `backgroundColor` prop. Both of these can't be used together.

So, try removing the `backgroundColor` prop. Then wrap the chart in a `View` and put the backgroundColor property inside the style of the View.

**Comment by NemyaNation** [comment]
I appreciate that, but honestly it doesn't look great because the horizontal rules are attached to the labels as you've previously stated.

This now brings the horizontal rules to the forefront as well.

I will postpone looking into this, if you ever decide to seperate styling for the y-axis labels / horizontal rules accompanying them; please let me know!

Right now I will just keep my current implementation.

**Comment by Abhinandan-Kushwaha** [comment]
Hi @NemyaNation 👋
Thanks for consistently persuading me to separate the styles for labels and rules. This was indeed very much required.
This is now implemented in version `1.0.21`🎉

Also, y axis labels and reference lines will have `zIndex` 1 by default.

**Comment by NemyaNation** [comment]
🎉🎉🎉🎉🎉
